tests/configurable_cluster.py

```diff
@@ -0,0 +1,188 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Docker related functions, constants and objects needed by product tests.
+
+Test writers should use this module for all of their cluster related needs.
+"""
+
+import fnmatch
+import os
+import paramiko
+from subprocess import check_call
+import tempfile
+import yaml
+
+from prestoadmin import main_dir
+from tests import user_password
+
+SSH_OPTIONS = ['-o', 'StrictHostKeyChecking=no']
+
+CONFIG_FILE_GLOB = r'*.yaml'
+DEFAULT_DOCKER_MOUNT_POINT = '/mnt/presto-admin'
+DIST_DIR = os.path.join(main_dir, 'tmp/installer')
+
+
+class ConfigurableCluster(object):
+    """Start/stop/control/query a cluster defined by a configuration file.
+
+    This class allows you to run the presto-admin product tests on a real
+    cluster.
+
+    """
+    def __init__(self, config_filename):
+        # see PyDoc for all_internal_hosts() for an explanation on the
+        # difference between an internal and regular host
+        with open(os.path.join(main_dir, config_filename)) as config_file:
+            config = yaml.load(config_file)
+
+        self.master = config['master']
+        self.slaves = config['slaves']
+        self.internal_master = config['internal_master']
```

Comment:
good point. I tried for a little while to make our tests not depend on master, slave1, slave2, and slave3, but it seemed like more effort than it was worth since you can just change /etc/hosts anyway...